TriblerGUI/widgets/trustpage/js/data_processor.js

```diff
@@ -0,0 +1,344 @@
+/**
+ * Process the GraphResponseData object into a GraphData object by collecting the result
+ * of a series of pure functions (with some exceptions).
+ *
+ * The resulting data structure is a GraphData object which is defined at the bottom
+ * of this file.
+ */
+
+/**
+ * Convert the GraphResponseData into GraphData by combining the result of pure
+ * functions.
+ *
+ * @param {GraphResponseData} response - The response data
+ * @returns {GraphData} - The resulting graph data object
+ */
+function processData(response) {
+
+    /**
+     * The reducer functions in their order of execution.
+     * @type {Reducer[]}
```

Comment:
A lot of developers will associate a reducer with [higher order function](https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29), so I would suggest to rename it to something like "converter".